lib/cocoapods-core/source.rb

```diff
@@ -24,6 +24,7 @@ class Source
     #
     def initialize(repo)
       @repo = Pathname(repo).expand_path
+      @name_to_versions_cache = {}
```

Comment:
Having said that, it might be correct for us to add code that invalidates when the repo is updated. Just because cocoapods installer calls things in the right order doesn't mean it should be designed as such.